azure-iot-device/azure/iot/device/provisioning/provisioning_device_client_factory.py

```diff
@@ -8,30 +8,27 @@
 communicate with Device Provisioning Service.
 """
 from .security.sk_security_client import SymmetricKeySecurityClient
-
 from .sk_provisioning_device_client import SymmetricKeyProvisioningDeviceClient
-from .transport.state_based_mqtt_provider import StateBasedMQTTProvider
+from azure.iot.device.provisioning.pipeline.pipeline_adapter import PipelineAdapter
 
 
-def create_from_security_client(provisioning_host, security_client, transport_choice):
+def create_from_security_client(security_client, transport_choice):
     """
     Creates different types of registration clients which can enable devices to communicate with Device Provisioning
     Service based on parameters passed.
-    :param provisioning_host: Host running the Device Provisioning Service. Can be found in the Azure portal in the
-    Overview tab as the string Global device endpoint
     :param security_client: Instance of Security client object which can be either of SymmetricKeySecurityClient,  TPMSecurtiyClient or X509SecurityClient
-    :param transport_choice: A string representing the transport the user wants
+    :param transport_choice: A string representing the pipeline the user wants
```

Comment:
I don't think this makes much semantic sense.

Reply:
iothub calls it a transport. are you referring to call it transport and go back to transport rather than pipeline ? the below line is from iothub
`:param transport: The transport that the client will use.`

---
In reply to: [287542349](https://github.com/Azure/azure-iot-sdk-python-preview/pull/97#discussion_r287542349) [](ancestors = 287542349)

Reply:
agreed that we should be consistent here, and I also think it's important to be consistent between languages.  I don't think any other SDKs use the term "protocol".  "transport_name" is what I put into iothub.  

---
In reply to: [287548505](https://github.com/Azure/azure-iot-sdk-python-preview/pull/97#discussion_r287548505) [](ancestors = 287548505,287542349)

Reply:
Yeah as long as we get it consistent between the two now, we can figure out what to do after. Let's go with `transport_name` for now.

---
In reply to: [287575351](https://github.com/Azure/azure-iot-sdk-python-preview/pull/97#discussion_r287575351) [](ancestors = 287575351,287548505,287542349)